org.eclipse.xtext.util/src/org/eclipse/xtext/util/PolymorphicDispatcher.java

```diff
@@ -353,7 +353,23 @@ protected RT handleAmbigousMethods(List<MethodDesc> result, Object... params) {
 		Collections.sort(cachedDescriptors, new Comparator<MethodDesc>() {
 			@Override
 			public int compare(MethodDesc o1, MethodDesc o2) {
-				return PolymorphicDispatcher.this.compare(o1, o2);
+				int compare = PolymorphicDispatcher.this.compare(o1, o2);
+				if (compare != 0) {
+					return compare;
+				}
+				Class<?>[] p1 = o1.getParameterTypes();
+				Class<?>[] p2 = o2.getParameterTypes();
+				int to = Math.min(p1.length, p2.length);
```

Comment:
You're probably just playing it safe, but doesn't `compare == 0` imply `p1.length == p2.length`?